api/lanes.py

```diff
@@ -374,32 +377,167 @@ def lane_for_other_languages(_db, exclude_languages):
+
+
+class RelatedBooksLane(LicensePoolBasedLane):
+    """A lane of Works all related to the Work of a particular LicensePool
+
+    Sublanes currently include a SeriesLane and a RecommendationLane"""
+
+    DISPLAY_NAME = "Related Books"
+
+    def __init__(self, _db, license_pool, full_name, display_name=None,
+                 mock_api=None):
+        sublanes = self._get_sublanes(_db, license_pool, mock_api=mock_api)
+        if not sublanes:
+            edition = license_pool.presentation_edition
+            raise ValueError(
+                "No related books for %s by %s" % (edition.title, edition.author)
+            )
+        super(RelatedBooksLane, self).__init__(
+            _db, license_pool, full_name,
+            display_name=display_name, sublanes=sublanes,
+            invisible=True
+        )
+
+    def _get_sublanes(self, _db, license_pool, mock_api=None):
+        sublanes = []
+
+        # Create a recommendations sublane.
+        try:
+            lane_name = "Recommendations for %s by %s" % (
+                license_pool.work.title, license_pool.work.author
+            )
+            sublanes.append(RecommendationLane(
+                _db, license_pool, lane_name, mock_api=mock_api
+            ))
+        except ValueError, e:
+            # NoveList isn't configured.
+            pass
+
+        # Create a series sublane.
+        series = license_pool.presentation_edition.series
+        if series:
+            lane_name = SeriesLane.lane_name_from_series_title(series)
+            sublanes.append(SeriesLane(_db, license_pool, lane_name))
+
+        return sublanes
+
+    def apply_filters(self, qu, *args, **kwargs):
+        # This lane is composed entirely of sublanes and
+        # should only be used to create groups feeds.
+        return None
 
+
+class SeriesLane(LicensePoolBasedLane):
+    """A lane of Works in a series based on a particular LicensePool"""
+
+    DISPLAY_NAME = "Other Books in this Series"
+
+    def apply_filters(self, qu, work_model=Work, *args, **kwargs):
+        edition = self.license_pool.presentation_edition
+        series = edition.series
+        if not series:
+            return None
+        qu = self.only_show_ready_deliverable_works(qu, work_model)
+
+        # Aliasing Edition here allows this query to function
+        # regardless of work_model and existing joins.
+        work_edition = aliased(Edition)
+        qu = qu.join(work_edition).filter(
+            work_edition.series==series,
+            work_edition.id!=edition.id
+        )
+        return qu
+
+    @classmethod
+    def lane_name_from_series_title(cls, series_title):
+        feed_title = "Other Books in "
+        if series_title[:3].lower() != 'the':
+            feed_title += "the "
+        feed_title += series_title
+        if series_title.lower().endswith(' series'):
+            return feed_title
+        return feed_title + ' series'
```

Comment:
I think it's a good idea. Does it matter that the book would be included in its own "also in this series" related feed?